scripts/npm-publish/npm-publish.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2025 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/perses/plugins/scripts/npm"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	tagNamePattern := regexp.MustCompile(`(?m)(.+)/v(\d.\d.\d)`)
+
+	tag := flag.String("tag", "", "Name of the tag")
+	flag.Parse()
+	tagSplitted := tagNamePattern.FindStringSubmatch(*tag)
+	if len(tagSplitted) != 3 {
+		logrus.Fatalf("Invalid tag name: %s", *tag)
+	}
+	pluginFolderName := tagSplitted[1]
+	version := tagSplitted[2]
+	// The manifest is hopefully uploaded by a previous task in the CI
+	// It should be available in the plugin folder
+	manifest, err := npm.ReadManifest(pluginFolderName)
+	if err != nil {
+		logrus.WithError(err).Fatalf("unable to read manifest file for plugin %s", pluginFolderName)
+	}
+	pluginName := manifest.Name
+	if execErr := exec.Command("npm", "publish", "--access", "public", filepath.Join(pluginFolderName, fmt.Sprintf("%s-%s.tar.gz", pluginName, version))).Run(); execErr != nil {
```

Comment:
this looks weird to me to publish on npm the `tar.gz` archive built. I am think about when you are importing the package. Usually it's done thanks to what is written in package.json giving the entrypoint and it is built with `tsc`.

How will it work with the static part ?

Reply:
oh ok, I didn't know that. Thanks for the explanation !